gph/src/ripser.h

```diff
@@ -280,24 +280,30 @@ struct compressed_distance_matrix {
 
     compressed_distance_matrix(std::vector<value_t>&& _distances)
         : distances(std::move(_distances)),
-          rows((1 + std::sqrt(1 + 8 * distances.size())) / 2)
+          rows((-1 + std::sqrt(1 + 8 * distances.size())) / 2)
     {
-        assert(distances.size() == size() * (size() - 1) / 2);
+        assert(distances.size() == size() * (size() + 1) / 2);
         init_rows();
     }
 
     template <typename DistanceMatrix>
     compressed_distance_matrix(const DistanceMatrix& mat)
-        : distances(mat.size() * (mat.size() - 1) / 2), rows(mat.size())
+        : distances(mat.size() * (mat.size() + 1) / 2), rows(mat.size())
     {
         init_rows();
 
-        for (size_t i = 1; i < size(); ++i)
-            for (size_t j = 0; j < i; ++j)
+        for (size_t i = 0; i < size(); ++i)
+            for (size_t j = 0; j <= i; ++j)
                 rows[i][j] = mat(i, j);
     }
 
-    value_t operator()(const index_t i, const index_t j) const;
+    inline value_t mat_tri_val(const index_t min, const index_t max) const;
+
+    value_t operator()(const index_t i, const index_t j) const
+    {
+        const auto& p = std::minmax(i, j);
```

Comment:
This is a bit tricky, this currently computes the min and max values, due to compiler optimization, no variable is created. It just helps to read the code.

Something interesting though, is that `std::minmax` with `SS2` optimization enable, allows to perform a branchless computation. In our case, it doesn't provide any performance enhancement/slowdown, but the code is ready that with specific compiler optimization enable. It could provide further speed-up.

Reply:
What do you think about adding a short version of this as an inline comment?  I fear that otherwise some of this wisdom will get lost!